service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ApplicantController.java

```diff
@@ -24,33 +31,70 @@
 
     @Autowired
     private MapperService mapperService;
+    private final ApplicantMigrationService applicantMigrationService;
+
+    @Autowired
+    public ApplicantController(MapperService mapperService,
+                               ApplicantMigrationService applicantMigrationService) {
+        this.mapperService = mapperService;
+        this.applicantMigrationService = applicantMigrationService;
+    }
+
+    @PostMapping("/about-to-start")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackrequest) {
+        CaseDetails caseDetails = callbackrequest.getCaseDetails();
+
+        return applicantMigrationService.setMigratedValue(caseDetails);
+    }
 
     @SuppressWarnings("unchecked")
     @PostMapping("/mid-event")
     public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestBody CallbackRequest callbackrequest) {
         CaseDetails caseDetails = callbackrequest.getCaseDetails();
+        ImmutableList.Builder<String> validationErrors = ImmutableList.builder();
 
-        Map<String, Object> applicantData = (Map<String, Object>) caseDetails.getData().get("applicant");
-        Applicant applicant = mapperService.mapObject(applicantData, Applicant.class);
-        List<String> validationErrors = new ArrayList<String>();
+        Map<String, Object> applicantData = (Map<String, Object>)
```

Comment:
Move it to `else` block. There is no reason to have it here since it is only used in `else` block

Reply:
👍